src/osd/OSD.h

```diff
@@ -1421,8 +1421,7 @@ class OSD : public Dispatcher,
 
     Spinlock sent_epoch_lock;
     epoch_t last_sent_epoch;
-    Spinlock received_map_lock;
-    epoch_t received_map_epoch; // largest epoch seen in MOSDMap from here
+    std::atomic<epoch_t> received_map_epoch; // largest epoch seen in MOSDMap from here
```

Comment:
Erm, apparently I've got it wrong how the function call relates to which parameters, but you probably get the idea: grab the current value, CAS with that value, loop if it failed and is still too low.